lib/explorer/shared.ex

```diff
@@ -286,38 +338,43 @@ defmodule Explorer.Shared do
   @doc """
   Downcasts lists of mixed numeric types (float and int) to float.
   """
-  def cast_numerics(list, type) when type == :numeric do
-    {cast_numerics_to_floats(list), {:f, 64}}
-  end
-
-  def cast_numerics(list, {:list, _} = dtype) do
+  def cast_numerics(list, dtype) do
     {cast_numerics_deep(list, dtype), cast_numeric_dtype_to_float(dtype)}
   end
 
-  def cast_numerics(list, type), do: {list, type}
+  defp cast_numerics_deep(list, {:struct, dtypes}) when is_list(list) do
+    Enum.map(list, fn item ->
+      Map.new(item, fn {field, inner_value} ->
+        inner_dtype = Map.fetch!(dtypes, to_string(field))
+        [casted_value] = cast_numerics_deep([inner_value], inner_dtype)
```

Comment:
This feels kinda inefficient (it's going to wrap/run through `Enum.map`/unwrap each value of the list of maps), but I was not too worried about it since I don't see `Series.from_list` as being the main way people will interact with this dtype. I feel like building a struct series inside a dataframe, or reading a struct series from an external file would be far more common.

Let me know if you disagree though.